### ISSUE TITLE
[HOLD for payment 2024-01-11] [$500] Wallet - The popup moves to the side if the page is resized

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#31706]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.2-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause-Internal Team
**Slack conversation:** @

## Action Performed:
Preconditions: The account has 1 or more bank account method

1. Go to URL https://staging.new.expensify.com/
2. Login with any account
3. Go to Settings -> Wallet -> Add bank account
4. Change the page size

## Expected Result:
The popup not moves

## Actual Result:
The popup moves

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

![Bug6287140_1700658137311!image__178_](https://github.com/Expensify/App/assets/115492554/cfc5beef-a756-44d7-9728-3b17071c6e94)

https://github.com/Expensify/App/assets/115492554/aa7ad83f-7d50-417c-beea-be2a51cb169d

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01bada80182536238f</li>
        <li>Upwork Job ID: 1727332858628177920</li>
        <li>Last Price Increase: 2023-11-29</li>
<li>Automatic offers: </li>
<ul>
<li>akinwale | Reviewer | 27955633</li>
<li>ishpaul777 | Contributor | 27955634</li>
</ul></li>
    </ul>
</details>

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Context menu is not updating it's position on dimension change
### What is the root cause of that problem?
This is a bug created by [this pr](https://github.com/Expensify/App/pull/27221) which migrates `PopoverReportActionContextMenu` from class to function component. It has broken the context menu position updating logic on dimension change.
### What changes do you think we should make in order to solve the problem?

We need to change `popoverAnchorPosition` from ref into a state
https://github.com/Expensify/App/blob/b4f001de6b3fe232a0f5aee5316b18557b740d86/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L28-L31
```
   const [popoverAnchorPosition, setPopoverAnchorPosition] = useState({
        horizontal: 0,
        vertical: 0,
    });
```
https://github.com/Expensify/App/blob/b4f001de6b3fe232a0f5aee5316b18557b740d86/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L89-L92C15
```
setPopoverAnchorPosition({
                horizontal: cursorRelativePosition.current.horizontal + x,
                vertical: cursorRelativePosition.current.vertical + y,
            });
```
https://github.com/Expensify/App/blob/b4f001de6b3fe232a0f5aee5316b18557b740d86/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L169-L172
```
setPopoverAnchorPosition({
                horizontal: nativeEvent.pageX,
                vertical: nativeEvent.pageY,
            });
```
https://github.com/Expensify/App/blob/b4f001de6b3fe232a0f5aee5316b18557b740d86/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L155
```
        contextMenuAnchorRef.current = contextMenuAnchor.current;

```
https://github.com/Expensify/App/blob/b4f001de6b3fe232a0f5aee5316b18557b740d86/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L164-L167
```
cursorRelativePosition.current = {
                horizontal: nativeEvent.pageX - x,
                vertical: nativeEvent.pageY - y,
            };
```
lastly change
https://github.com/Expensify/App/blob/b4f001de6b3fe232a0f5aee5316b18557b740d86/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L293
```
                anchorPosition={popoverAnchorPosition}

```
### What alternative solutions did you explore? (Optional)

**Comment by Victor-Nyagudi** [comment]
Might be a good idea to hold this issue until the following is merged or maybe combine them as they both deal with the pop up being misaligned when the screen is resized. 

- https://github.com/Expensify/App/issues/29729

**Comment by FitseTLT** [comment]
@Victor-Nyagudi I disagree with you. By the way, the title of this issue brings misunderstanding that it seems it is related with the delete modal on Wallet but the case is as commented in my [proposal](https://github.com/Expensify/App/issues/31706#issuecomment-1823347399) it is a regression caused when migrating `PopoverReportActionContextMenu` from class component to functional component and the bug is breaking the popover mechanism to reposition itself on dimension change and now every context menu we have is broken (in terms of repositioning) and this bug should be fixed ASAP.
Look as this video 👍 

https://github.com/Expensify/App/assets/38649957/c1533975-6141-4b82-9691-c75a158660f6

**Comment by FitseTLT** [comment]
> @FitseTLT unless i am missing something here, this particular issue aims to solve the issue for wallet screen popover not the context menu for chat it should be treated as a seperate issue (if it is a issue as far as i guess this is expected for context menu and this does not create any ui inconsistency) you can raise this on slack channel if you are sure this is a issue needs to fixed and there's no GH handling that already
Thanks @ishpaul777 you're right. Because I couldn't add a bank account (via Plaid) when I reproduced exact bug with PopoverReportActionContextMenu I thought we were using the same popover 🙏 in the wallet page.
@akinwale @kevinksullivan Can U help create a new issue to deal with the `PopoverReportActionContextMenu` repositioning issue commented in my [proposal](https://github.com/Expensify/App/issues/31706#issuecomment-1823347399) and indicated below in the videos. It is even more important issue than this one b/c `PopoverReportActionContextMenu` is far more commonly used popover menu.

https://github.com/Expensify/App/assets/38649957/87749c60-57b5-416b-8d86-1f34a41281a1


https://github.com/Expensify/App/assets/38649957/c1533975-6141-4b82-9691-c75a158660f6

**Comment by FitseTLT** [comment]
@kevinksullivan But I am not in that #expensify-bugs slack channel. I think Expensify has stopped accepting bugs from Contributors but it is an important bug. 😕

**Comment by kevinksullivan** [comment]
Still looking for proposals

**Comment by akinwale** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [x]  [@akinwale] The PR that introduced the bug has been identified. Link to the PR:
> * [x]  [@akinwale] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
> * [x]  [@akinwale] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Not a regression. The popup is initially positioned based on the coordinates of the clicked UI element, and there was no event handler implemented to reposition after the window / viewport is resized.

> * [x]  [@akinwale] Determine if we should create a regression test for this bug.
> * [x]  [@akinwale] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Regression test steps**
* Open the Expensify URL in the browser and login
* Navigate Settings > Wallet
* Add a bank account if there are none added
* Click on the bank account to display the context popup
* Resize the viewport (eg, open and close the browser developer tools pane)
* Verify that the popup is repositioned properly after the window is resized

Do we agree 👍 or 👎?

**Comment by francoisl** [comment]
Internal issue to add the regression test created ([360448](https://github.com/Expensify/Expensify/issues/360448)), @kevinksullivan we'll just need to handle payment and close.

**Comment by akinwale** [comment]
@miljakljajic bump for payment based on https://github.com/Expensify/App/issues/31706#issuecomment-1887598363.

Thanks.

**Comment by miljakljajic** [comment]
@kevinksullivan seems like we're accidentally assigning another BZ member when one is already assigned. 

@akinwale I'll issue payment now!

**Comment by akinwale** [comment]
> @akinwale I'll issue payment now!

Got it. Thanks!
